pkg/instrumentation/podmutator.go

```diff
@@ -111,3 +114,10 @@ func (pm *instPodMutator) selectInstrumentationInstanceFromNamespace(ctx context
 		return otelInsts.Items[0], nil
 	}
 }
+
+func (pm *instPodMutator) applyDefaults(otelinst v1alpha1.Instrumentation) v1alpha1.Instrumentation {
+	if otelinst.Spec.Java.Image == "" {
```

Comment:
This should be part of the Defaulter.

https://github.com/open-telemetry/opentelemetry-operator/blob/de6219f576b1af2e88c9119452ae8ef1db67cb7a/apis/collector/v1alpha1/opentelemetrycollector_webhook.go#L39-L55

Reply:
Collector image is not handled in the defaulter either. 

That said I might agree with you (I need to test the defaulter, when it comes to updates).  I have this WIP PR https://github.com/open-telemetry/opentelemetry-operator/pull/517 that might change this to use the defaulter.

Reply:
The collector image was probably handled before we implemented the defaulter.

Reply:
I would like to get this merged and then change just remove this function in the #517 PR.